src/Altinn.App.Core/Helpers/ObjectUtils.cs

```diff
@@ -137,14 +138,31 @@ public static void PrepareModelForXmlStorage(object model, int depth = 64)
                 SetToDefaultIfShouldSerializeFalse(model, prop, methodInfos);
 
                 // Set string properties with [XmlText] attribute to null if they are empty or whitespace
-                if (
-                    value is string s
-                    && string.IsNullOrWhiteSpace(s)
-                    && prop.GetCustomAttribute<XmlTextAttribute>() is not null
-                )
+                if (value is string s)
                 {
-                    // Ensure empty strings are set to null
-                    prop.SetValue(model, null);
+                    if (string.IsNullOrWhiteSpace(s))
+                    {
+                        if (prop.GetCustomAttribute<XmlTextAttribute>() is not null)
+                        {
+                            // Ensure empty strings are set to null
+                            prop.SetValue(model, null);
+                        }
+                    }
+                    else
+                    {
+                        var xmlRegex = XmlInvalidCharsRegex();
+                        if (prop.SetMethod is not null)
+                        {
+                            // Remove invalid xml characters
+                            prop.SetValue(model, xmlRegex.Replace(s, "�"));
```

Comment:
I've had a few instances where editors mess up unicode, so I tend to prefere to use the unicode escapes in string literals. Especially here when it is the error character.

```suggestion
                            prop.SetValue(model, xmlRegex.Replace(s, "\uFFFD")); // \uFFFD is the unicode replacement character �
```